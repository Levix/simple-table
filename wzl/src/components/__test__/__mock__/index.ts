import type { TableColumns, TableDataSourceType } from '../../types';
import { useDataSource } from "../../../composables/useDataSource";
import { useColumns } from "../../../composables/useColumns";
import { ref } from 'vue';

const dataSource: TableDataSourceType[] = []
for (let index = 0; index < 10; index++) {
    dataSource.push({
        key: index,
        name: `Edrward ${index}`,
        age: 18 + index,
        address: `London Park no. ${index}`,
        tags: ['nice', 'developer'],
    })
}
// 表格源数据
export { dataSource };
// 表格列配置
export const columns: TableColumns[] = [
    {
		title: "Name",
		key: "name",
	},
	{
		title: "Age",
		key: "age",
		width: 100,
		align:'center',
		sortable: {
			orderBy: "DESC",
			sorter: (curr, next) => next - curr,
		},
	},
	{
		title: "Address",
		key: "address",
	},
	{
		title: "Tags",
		key: "tags",
	},
];

export const sortIconObj = {
    ascTriangle: "▲",
    descTriangle: "▼",
    normalTriangle: "■",
}

export const a_columns: TableColumns[] = [
    {
        title: "Col_1",
        key: "Col__1",
    },
    {
        title: "Col_2",
        key: "Col__2",
        sortable: {
            orderBy: "asc"
        }
    },
    {
        title: "Col_3",
        key: "Col__3",
    },
    {
        title: "Col_4",
        key: "Col__4",
    },
]
export const d_columns: TableColumns[] = [
    {
        title: "Col_1",
        key: "Col__1",
    },
    {
        title: "Col_2",
        key: "Col__2",
        sortable: {
            orderBy: "desc"
        }
    },
    {
        title: "Col_3",
        key: "Col__3",
    },
    {
        title: "Col_4",
        key: "Col__4",
    },
]

// 处理后的配置
export const { newColumns } = useColumns(columns);
// 处理后的数据
export const { sortTableData } = useDataSource(ref(dataSource), newColumns);
// body配置
export const bodyOptions = {
    sortData: sortTableData.value,
    columns: newColumns.value,
    sortOption: {
        key: 1,
        sortStatus: 'asc',
    },
    pageSize: 10,
    curPageNumber: 1,
}

// foot配置
export const footOptions = {
    sortData: sortTableData.value,
    pageSize: 4,
}

export const newFootOptions = {
    sortData: sortTableData.value,
    pageSize: 8,
}