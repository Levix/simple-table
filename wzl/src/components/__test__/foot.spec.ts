import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import WuFoot from '../Foot';
import { footOptions, newFootOptions } from './__mock__/index';

describe('WuFoot', () => {
    const wrapper = mount(WuFoot, {
        props: {
            footOptions: footOptions,
        }
    })

    it('打个快照 WuFoot', () => {
        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.exists()).toBe(true);
    })

    it('组件是否正常渲染，参数是否正确', async () => {
        expect(wrapper.find('tfoot').exists()).toBe(true);

        expect(wrapper.findAll('button')).toHaveLength(Math.ceil(footOptions.sortData.length / footOptions.pageSize) + 2);

        await wrapper.setProps({
            footOptions: newFootOptions
        })

        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.findAll('button')).toHaveLength(Math.ceil(newFootOptions.sortData.length / newFootOptions.pageSize) + 2);

        // 点击事件是否触发
        const prevButton = wrapper.find('.wu_foot__prev');
        // 初始禁用状态
        expect(prevButton.classes('prev_disabled')).toBe(true);

        const nextButton = wrapper.find('.wu_foot__next');
        // 初始不禁用,因为数据不只是一页
        expect(nextButton.classes('next_disabled')).toBe(false);

        // 点击下一页,上一页将解除禁用
        await nextButton.trigger('click');
        expect(prevButton.classes('prev_disabled')).toBe(false);
        
        // 点击上一页,上一页禁用,页签按钮进入按钮1,带actived类;
        await prevButton.trigger('click');
        expect(prevButton.classes('prev_disabled')).toBe(true);
        expect(wrapper.findAll('button').at(1)?.classes('actived')).toBe(true);
    })
})