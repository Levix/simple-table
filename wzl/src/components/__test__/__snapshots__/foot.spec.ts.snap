// Vitest Snapshot v1

exports[`WuFoot > 打个快照 WuFoot 1`] = `
"<tfoot class=\\"wu_foot\\">
  <td colspan=\\"4\\"><button class=\\"wu_foot__prev prev_disabled\\" disabled=\\"\\">前一页</button><button class=\\"actived\\">1</button><button class=\\"\\">2</button><button class=\\"\\">3</button><button class=\\"wu_foot__next\\">后一页</button>
    <div class=\\"wu_foot__box\\"><span>当前：第1页</span><br><span>4条 / 页</span><br><span>总共：10条</span></div>
  </td>
</tfoot>"
`;

exports[`WuFoot > 组件是否正常渲染，参数是否正确 1`] = `
"<tfoot class=\\"wu_foot\\">
  <td colspan=\\"4\\"><button class=\\"wu_foot__prev prev_disabled\\" disabled=\\"\\">前一页</button><button class=\\"actived\\">1</button><button class=\\"\\">2</button><button class=\\"wu_foot__next\\">后一页</button>
    <div class=\\"wu_foot__box\\"><span>当前：第1页</span><br><span>8条 / 页</span><br><span>总共：10条</span></div>
  </td>
</tfoot>"
`;
